### PR TITLE
Prac 07 feedback

### PR DESCRIPTION
**Pull request by luke185**
@Lucas-obrien can you look over my code and provide useful feedback, in particular can you check to see if my names are good

### CODE REVIEW

File: prac_07/myguitars.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    name = input('Name: ')
+    while name != '':
+        year = int(input('Year: '))
+        cost = float(input('Cost: '))
+        guitars.append(Guitar(name, year, cost))
+        name = input('Name: ')
+
+    display_guitars(guitars)
+
+    with open('guitars.csv', 'w') as out_file:
```
Comment (Lucas-obrien) [inline prac_07/myguitars.py:30]: Consider making load and save functions to keep main looking consistent

File: prac_07/project_management.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            projects.append(Project(line[0], line[1], int(line[2]), float(line[3]), int(line[4])))
+    return projects
+
+
+def display_projects(projects):
+    """display all projects grouped by completion status"""
+    projects.sort()
+    print('Incomplete projects:  ')
+    for project in projects:
+        if project.completion_percentage < 100:
```
Comment (Lucas-obrien) [inline prac_07/project_management.py:57]: Could you make this a class method instead?

File: prac_07/project_management.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    """update a chosen project"""
+    for i, project in enumerate(projects):
+        print(f"{i} {project}")
+    choice = int(input('Project choice: '))
+    print(projects[choice])
+    new_percentage = int(input('New percentage: '))
+    projects[choice].completion_percentage = new_percentage
+    new_priority = int(input('New priority: '))
+    projects[choice].priority = new_priority
+    return projects
```
Comment (Lucas-obrien) [inline prac_07/project_management.py]: Could you just alter the list in the function and not return it to save writing over the whole list?

File: prac_07/project_management.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            filter_date = input('Show projects that start after date (dd/mm/yy): ')
+            filter_date = datetime.datetime.strptime(filter_date, "%d/%m/%Y").date()
+            filter_by_date(projects, filter_date)
+        elif choice == 'A':
+            add_new_project(projects)
+        elif choice == 'U':
+            projects = update_project(projects)
+        else:
+            print('Invalid Input')
+        choice = input('> ').upper()
```
Comment (Lucas-obrien) [inline prac_07/project_management.py:37]: Maybe reprint the menu here, it's difficult to tell where you are in the code after you've selected a menu option

File: prac_07/project_management.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    with open(filename, 'w') as out_file:
+        print("Name\tStart Date\tPriority\tCost Estimate\tCompletion Percentage", file=out_file)
+        for project in projects:
+            print(f"{project.name}\t{project.start_date}\t{project.priority}\t{project.cost_estimate}\t"
+                  f"{project.completion_percentage}", file=out_file)
+
+
+def filter_by_date(projects, filter_date):
+    """only show projects after a given date and sort by date"""
+    for project in projects:
```
Comment (Lucas-obrien) [inline prac_07/project_management.py]: I can see what you're going for here, but it converts the objects date string to a date value, which alters your display later on in the function. Consider using the __gt__ method in your class and just converting the string date for the comparison

File: prac_07/project_management.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
+"""
+CP1404 Prac 07 - project_management.py
+Estimated time: 60m
+Actual time:
```
Comment (Lucas-obrien) [inline prac_07/project_management.py:4]: This was a rough practical week

**Review by Lucas-obrien — COMMENTED** [review]
Hey Luke, just a few things mainly in the project.py, this prac was huge so it's 100% understandable that full functionality isn't there.

Main thing to watch out for is your docstrings, there are a few missing and some need the formatting updated

File: prac_07/project_management.py (hunk 6 above)
Comment (luke185) [reply prac_07/project_management.py:4]: Great feedback, caught a few things I didn't think of. Most of the issues were due to me having rush the completion of the prac. Thanks for the feedback : )

File: prac_07/project_management.py (hunk 5 above)
Comment (luke185) [reply prac_07/project_management.py]: thanks this one was a big help, didn't think to use the gt method

File: prac_07/languages.csv [hunk 7]
```diff
@@ -0,0 +1,7 @@
+Language,Typing,Reflection,Year
```
Comment (lindsaymarkward) [inline prac_07/languages.csv:1]: 4 headings for 5 fields

File: prac_07/project.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+CP1404 Prac 07 - project.py
+"""
+import datetime
+
+
+class Project:
+    def __init__(self, name='', start_date='', priority=0, cost_estimate=0.0, completion_percentage=0):
+        """Initialise a project"""
+        self.name = name
+        self.start_date = datetime.datetime.strptime(start_date, "%d/%m/%Y").date()
```
Comment (lindsaymarkward) [inline prac_07/project.py:11]: This shouldn't be Project's job. Do you convert the priority to an int here?

File: prac_07/project_management.py [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    print('Thanks : )')
+
+
+def load_projects(filename):
+    """load a projects file and format it into a list of project objects"""
+    projects = []
+    with open(filename, 'r') as in_file:
+        in_file.readline()
+        for line in in_file:
+            line = line.strip().split('\t')
```
Comment (lindsaymarkward) [inline prac_07/project_management.py:47]: line is not a good name for a list

**Pull request merged**